src/proto.rs

```diff
@@ -91,3 +93,11 @@ impl ResponseData {
         unsafe { mem::transmute(&self.response) }
     }
 }
+
+pub type ImapTls = Framed<TlsStream<TcpStream>, ImapCodec>;
+
+impl ImapTransport for ImapTls {}
+
+pub trait ImapTransport: futures::Stream<Item = ResponseData, Error = io::Error> +
+                         futures::Sink<SinkItem = Request, SinkError = io::Error> {
+}
```

Comment:
If only we had trait aliases :(

Reply:
Yeah, tried that first. Oh, well, at least this works...

Reply:
Interesting remark Jon. We do not have [trait aliases](https://github.com/rust-lang/rfcs/blob/master/text/1733-trait-alias.md) yet, but I concur that this works. I don’t see trait aliases being a boon here but perhaps I’m missing something.